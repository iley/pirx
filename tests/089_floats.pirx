func main(): int {
  // Test float64 operations
  var a: float64;
  var b: float64;
  var result: float64;
  
  a = 2.5;
  b = 1.5;
  
  // Addition
  result = a + b;
  printf("add: %.1f\n", result);
  
  // Subtraction
  result = a - b;
  printf("sub: %.1f\n", result);
  
  // Multiplication
  result = a * b;
  printf("mul: %.1f\n", result);
  
  // Division
  result = a / b;
  printf("div: %.1f\n", result);
  
  // Test with literals
  result = 0.1 + 0.2;
  printf("literals: %.1f\n", result);
  
  // Test more operations
  result = 10.0 - 3.5;
  printf("sub2: %.1f\n", result);
  
  result = 1.5 * 2.0;
  printf("mul2: %.1f\n", result);
  
  result = 7.5 / 2.5;
  printf("div2: %.1f\n", result);
  
  return 0;
}
