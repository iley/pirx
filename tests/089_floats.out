add: 4.0
sub: 1.0
mul: 3.8
div: 1.7
literals: 0.3
sub2: 6.5
mul2: 3.0
div2: 3.0
